
================== CRASH DETECTED ==================
Time: 2025-09-09 00:04:12
Runtime: 1.55 seconds
Exception Type: AttributeError
Exception Value: 'DistanceBasedCachingManager' object has no attribute '_mutex'

Traceback:
Traceback (most recent call last):
  File "C:\dcwidget\ui_main.py", line 1195, in <module>
    widget = MainWidget(settings)
             ^^^^^^^^^^^^^^^^^^^^
  File "C:\dcwidget\ui_main.py", line 146, in __init__
    self.initUI()
  File "C:\dcwidget\ui_main.py", line 519, in initUI
    self.set_current_date(self.current_date, is_initial=True)
  File "C:\dcwidget\ui_main.py", line 595, in set_current_date
    self.refresh_current_view()
  File "C:\dcwidget\ui_main.py", line 679, in refresh_current_view
    current_widget.refresh()
  File "C:\dcwidget\views\month_view.py", line 363, in refresh
    self._last_event_hash = self._calculate_events_hash()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\dcwidget\views\month_view.py", line 190, in _calculate_events_hash
    all_events = self.data_manager.get_events(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\dcwidget\data_manager.py", line 825, in get_events
    self.notify_date_changed(datetime.date(year, month, 1))
  File "C:\dcwidget\data_manager.py", line 687, in notify_date_changed
    self.caching_manager.request_caching_around(new_date.year, new_date.month)
  File "C:\dcwidget\data_manager.py", line 325, in request_caching_around
    with QMutexLocker(self._mutex):
                      ^^^^^^^^^^^
AttributeError: 'DistanceBasedCachingManager' object has no attribute '_mutex'

================== END CRASH INFO ==================
